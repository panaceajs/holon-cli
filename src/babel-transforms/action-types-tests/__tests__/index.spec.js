const pluginTester = require('babel-plugin-tester');
const actionTypesTests = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: actionTypesTests,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should change nothing when there is nothing to change.',
      fixture: '__fixtures__/missing-describe-call.js',
      snapshot: false
    },
    {
      title: 'it should add tests for action types',
      fixture: '__fixtures__/with-describe-call.js',
      pluginOptions: { actionTypes: { doSomething: 'doSomething' } }
    },
    {
      title: 'it should complete tests for action types',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        actionTypes: {
          existingAction: 'existingAction',
          nonExistingAction: 'nonExistingAction'
        }
      }
    }
  ]
});
