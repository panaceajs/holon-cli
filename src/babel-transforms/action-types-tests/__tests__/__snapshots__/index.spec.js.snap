// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-action-types-tests it should add tests for action types: it should add tests for action types 1`] = `
"
describe('action types', () => {});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`root\` action types', () => {
  it('should export \`DO_SOMETHING\` type', () => {
    expect(DO_SOMETHING).toBe('DO_SOMETHING');
  });
});
"
`;

exports[`holon-action-types-tests it should complete tests for action types: it should complete tests for action types 1`] = `
"
describe('action types', () => {
  it('should export \\\\\`EXISTING_ACTION\\\\\` type', () => {
    // existing
    expect(EXISTING_ACTION).toBe('EXISTING_ACTION');
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`root\` action types', () => {
  it('should export \\\\\`EXISTING_ACTION\\\\\` type', () => {
    // existing
    expect(EXISTING_ACTION).toBe('EXISTING_ACTION');
  });
  it('should export \`NON_EXISTING_ACTION\` type', () => {
    expect(NON_EXISTING_ACTION).toBe('NON_EXISTING_ACTION');
  });
});
"
`;
